src/translator/processed_request_sentence.cpp

```diff
@@ -0,0 +1,102 @@
+#include "processed_request_sentence.h"
+
+#include <cstring>
+#include <iostream>
+
+namespace marian {
+namespace bergamot {
+
+namespace {
+
+// Generic write/read from pointers
+
+// write uses ostream, to get bytes/blob use with ostringstream(::binary)
+template <class T>
+void write(std::ostream &out, const T *data, size_t num = 1) {
```

Comment:
I think you're using ostream in performance-critical code?  ostream is crazy slow.

Reply:
Also this method name is bad, since there is a C write function.